keychain-sdk/tx_client_pool.go

```diff
@@ -0,0 +1,124 @@
+package keychain
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	sdkTypes "github.com/cosmos/cosmos-sdk/types"
+	client "github.com/warden-protocol/wardenprotocol/go-client"
+	"google.golang.org/grpc/connectivity"
+)
+
+type AppClient struct {
```

Comment:
Let’s avoid using the term “app” which is overloaded (in this case we’re connecting to a warden node, it can be confused with the keychain app itself), plus avoid exporting it because as far as I understand the use won’t initialize this manually

```suggestion
type wardenClient struct {
```

Reply:
+